### PR TITLE
Improve info.py script

### PR DESCRIPTION
**Pull request by MaciejWanat**
Hey, I've been struggling with issue of preview not displaying correctly and wrote myself a script before I noticed that you already did that :)

Your code is great but I've done 2 things which I consider improvements when dealing with large files:

1. Analyze only last 400 lines, not all of them - since the info is always in the footer we don't need to read and compare through whole gcode (especially that they can get large)
2. Support days. In current implementation days are ignored, so print time wont display correctly if its longer than 1 day.

If you find that an improvement, feel free to merge in!

### CODE REVIEW

**Comment by suchmememanyskill** [comment]
Good suggestions, but i don't agree with how they're implemented. You're looping multiple times over those 400 lines now. I'll look at this more in depth tonight (+/-8h)

**Comment by MaciejWanat** [comment]
True! Reiterating 400 lines should still be faster than checking whole file once (gcodes are rarely shorter than 1200 lines), and I dont think it's very costly to run the search on 400 lines 3 times, I think this is neglectable. On the other hand searching once through file that can easily be over 1 milion rows will be much slower.

But we can switch this to use `startswith`, of course.

**Comment by MaciejWanat** [comment]
Out of curiosity I ran a test on 4,9 milion rows file. 

Original: real    0m2,581s
Mine: real    0m2,187s

So its quite neglactable either way, it seems (not sure how would it behave on weaker CPU)

**Comment by suchmememanyskill** [comment]
I don't entirely understand why the entire file should be replaced, rather than just
```diff
diff --git a/img/info.py b/img/info.py
index 2e0a52b..34ca47e 100644
--- a/img/info.py
+++ b/img/info.py
@@ -9,12 +9,14 @@ filament_used_mm = 0
 time_s = 0
 layer_height = 0
 
-for line in data:
+for line in data[-400:]:
     if line.startswith("; filament used [mm] = "):
         filament_used_mm = float(line[23:])
     if line.startswith("; estimated printing time (normal mode) = "):
         line_times = [x.strip() for x in line[42:].split(" ")]
         for time in line_times:
+            if time.endswith("d"):
+                time_s += int(time[:-1]) * 3600 * 24
             if time.endswith("h"):
                 time_s += int(time[:-1]) * 3600
             elif time.endswith("m"):
```

**Comment by MaciejWanat** [comment]
It doesn't need to be replaced, of course - I just proposed my implementation because I wrote it whole unaware of existence of yours. Editing yours works too of course - if you want to make indication of changes cleaner, we can go this way. 

Pushed code with following your suggestion.

**Comment by suchmememanyskill** [comment]
I'd prefer to use mine, yea. I am not that good with regex and wouldn't be able to reproduce what you wrote. I have some comments, though. Second

File: img/info.py [hunk 1]
```diff
@@ -8,13 +8,16 @@
 filament_used_mm = 0
 time_s = 0
 layer_height = 0
+bottomLinesAnalyzedAmount = 400
```
Comment (suchmememanyskill) [inline img/info.py:11]: Can you make this naming snake case so it matches the other variable names?

File: img/info.py [hunk 2]
```diff
@@ -34,6 +37,7 @@
     f";TIME:{time_s:.2f}",
     f";Filament used:{filament_used_mm/1000:.5f}m",
     f";Layer height:{layer_height}",
+    "';Generated by EnderV3SEPrusaPostScript",
```
Comment (suchmememanyskill) [inline img/info.py:40]: Can you instead make this a link to this repo, [vaguely like the config bundle](https://github.com/suchmememanyskill/PrusaSlicer-Ender3-v3-SE-Config/blob/main/generate_config_bundle.py#L4)?

**Review by suchmememanyskill — CHANGES_REQUESTED** [review]

File: img/info.py (hunk 1 above)
Comment (MaciejWanat) [reply img/info.py:11]: Sorry, .net habits ;)

**Comment by MaciejWanat** [comment]
Pushed the changes requested.
By the way time on the same file is `real    0m1,955s` so hey, dropping regexps spared us 0,19s :D

File: img/info.py [hunk 3]
```diff
@@ -35,7 +35,7 @@
 
 extra_data = [
     ";FLAVOR:Marlin",
-    f";TIME:{time_s:.2f}",
```
Comment (suchmememanyskill) [inline img/info.py:34]: Does dropping the formatter still make it work on the screen? The output generated by creality slicer has 2 decimal points

Comment (suchmememanyskill) [reply img/info.py:34]: This is one thing i cannot test myself; i don't run stock firmware anymore

**Comment by MaciejWanat** [comment]
One more change, your code is giving `;TIME:134173.00` rather than `;TIME:134173` - second one is expected I think. Running long print so cannot check right now, but that's what cura does.

File: img/info.py (hunk 3 above)
Comment (MaciejWanat) [reply img/info.py:34]: Yeah, cura generates this output:

```;FLAVOR:Marlin
;TIME:9994
;Filament used: 9.3909m
;Layer height: 0.2
;MINX:66.534
;MINY:68.118
;MINZ:0.2
;MAXX:153.365
;MAXY:151.896
;MAXZ:70.6
;TARGET_MACHINE.NAME:Creality Ender-3 V3 SE
;Generated with Cura_SteamEngine 5.6.0```

and this is what I was looking for. If creality does that with decimal points we can withdraw this last commit of mine, probably both could work

**Comment by suchmememanyskill** [comment]
Had a friend try it, it works with and without the decimal digits

**Review by suchmememanyskill — APPROVED** [review]
Ready for merge. Tell me when you're ready

**Comment by MaciejWanat** [comment]
Great! Lets go :) Happy to contribute.

**Comment by suchmememanyskill** [comment]
@MaciejWanat i've rewritten it in .NET so i can use native AOT and just generate an .exe on windows (probably easier for end users), it's uh slightly faster lol 
image

**Comment by MaciejWanat** [comment]
Not sure if it's easier. I think most people who use scripts are used to python, at least when I was researching vast majority was written in python. So I'd probably attach both of these files with instructions for both.

**Comment by suchmememanyskill** [comment]
@MaciejWanat on macos/linux, it'd be easier to use python yeah. But windows it's easier to use a standalone exec as you don't have to use python (i don't think a lot of non-programmers have python installed)

**Pull request merged**